ui/Screen/RegisterUser/PickHandleStep.svelte

```diff
@@ -0,0 +1,108 @@
+<script>
+  import validatejs from "validate.js";
+  import { Button, Flex, Input } from "../../DesignSystem/Primitive";
+
+  import { pop } from "svelte-spa-router";
+  import { gql } from "apollo-boost";
+  import { getClient, query } from "svelte-apollo";
+
+  export let onNextStep = null;
+
+  const nextStep = () => {
+    if (!validatejs.isEmpty(validations)) {
+      return;
+    }
+    onNextStep();
+  };
+
+  export let handle = "";
+
+  let validating = false;
+  let validations = false;
+  let timeout = null;
+  let delay = 0;
+  const client = getClient();
+
+  const GET_USER = gql`
+    query Query($handle: ID!) {
+      user(handle: $handle)
+    }
+  `;
+
+  const validateHandleAvailability = async () => {
+    try {
+      const response = await query(client, {
+        query: GET_USER,
+        variables: { handle: handle }
+      });
+      const result = await response.result();
+      if (await result.data.user) {
+        validations = { handle: ["Handle already taken"] };
+      }
+    } catch (error) {
+      validations = { handle: [error] };
+    }
+  };
+
+  validatejs.options = {
+    fullMessages: false
+  };
+
+  const VALID_NAME_MATCH = new RegExp("^[a-z0-9][a-z0-9_-]+$", "i");
```

Comment:
This match is not according to the spec in https://github.com/radicle-dev/registry-spec/ - skeptical if we should maintain this in the UI at all as it easily gets out of sync. Can we defer to the proxy to be authoritative on this?

Reply:
Issue to move validations into proxy: #219

The specifications in https://www.notion.so/monadic/Basic-User-Identity-0688a802e65040778a459892d6922919 also diverge from the `registry-spec`. I would leave the regexp as is for now.

Reply:
The regexp in the Notion document is not in regards to the user handle input allowed on chain, but the restrictions for handles that are part of the identity metadata in CoCo. The spec is authoritative for all Registry concerns.

It's fine to leave it for now and capture the follow-up in #219.